### PR TITLE
Fix timeline drawer measurement visualization

### LINKED ISSUE #5361: timeline_drawer measurement bugs
There are some bugs drawing measurements in the circuit timeline drawer. Taking a simple circuit and a backend to schedule on:

```py
from qiskit import QuantumCircuit, transpile
from qiskit.test.mock import FakeParis
from qiskit.visualization import timeline_drawer

qc = QuantumCircuit(2, 2)
qc.cx(0, 1)

backend = FakeParis()
```

The following scenarios have various errors...

1- One measurement with show_clbits=False
```py
qc.measure(0, 0)
new_qc = transpile(qc, backend, scheduling_method='alap')
timeline_drawer(new_qc, show_clbits=False)
```
`KeyError: Clbit(ClassicalRegister(2, 'c'), 0)`

2- One measurement with show_clbits=True
```py
qc.measure(0, 0)
new_qc = transpile(qc, backend, initial_layout=[25, 26], scheduling_method='alap')
timeline_drawer(new_qc, show_clbits=True)
```

The clbit measure seems off and starts earlier than the qubit measure.

image

3- Two measurement with show_clbits=False
```py
qc.measure(0, 0)
qc.measure(1, 1)
new_qc = transpile(qc, backend, initial_layout=[25, 26], scheduling_method='alap')
timeline_drawer(new_qc, show_clbits=False)
```

`TypeError: '<=' not supported between instances of 'float' and 'NoneType'`

4- Two measurements with show_clbits=True
```py
qc.measure(0, 0)
qc.measure(1, 1)
new_qc = transpile(qc, backend, initial_layout=[25, 26], scheduling_method='alap')
timeline_drawer(new_qc, show_clbits=True)
```

`AttributeError: 'list' object has no attribute 'repeat'`

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
fix #5361 

### Details and comments
Fix bit link visualization. Currently the drawer tries to visualize the bit link between qreg and creg even though the creg is disabled. This causes an error because the coordinate of the creg is not initialized. In this PR, visible check is moved to correct place so that the link is also disabled. Unittests are also updated.

Box position bug of cregs is not fixed with this PR.

### CODE REVIEW

File: qiskit/visualization/timeline/interface.py [hunk 1]
```diff
@@ -356,6 +356,13 @@ def draw(program: circuit.QuantumCircuit,
     if show_delays is not None:
         temp_style['formatter.control.show_delays'] = show_delays
 
+    # currently clbits starts at t = 0.
+    # TODO fix this
+    if temp_style['formatter.control.show_clbits']:
+        import warnings
+        warnings.warn('Start time of measure instruction on Clbits is not correctly parsed. '
```
Comment (taalexander) [inline qiskit/visualization/timeline/interface.py:363]:
```suggestion
        warnings.warn('Start time of the measure instruction on Clbits is not correctly parsed. '
```

**Review by taalexander — CHANGES_REQUESTED** [review]
The warning is being triggered in docs building.

**Comment by nkanazawa1989** [comment]
Is there any good way of avoiding this? We can add this warning but current creg output may confuse the users (because measurement on creg always starts from wrong position because creg doesn't have time information).

**Comment by taalexander** [comment]
> Is there any good way of avoiding this? We can remove this warning but current creg output may confuse the users (because measurement on creg always starts from wrong position because creg doesn't have time information). Another option is removing debugging stylesheet from docstring, but I want to keep this.

Why is there a warning being emitted into the documentation?

**Comment by nkanazawa1989** [comment]
Because in the api docs I showed appearance of timeline images with three different preset stylesheets, namely `IQXSimple`, `IQXStandard`, `IQXDebugging`. Currently `show_clbits=True` generates a wrong image but `IQXDebugging` style should show whatever the drawer can according to its stylesheet policy. We should fix this bug at some point but I need more discussion.

**Comment by taalexander** [comment]
I think right now it will be required to remove whatever is causing the warning in the documentation.

**Comment by nkanazawa1989** [comment]
close because this is addressed in #5662

**Comment by nkanazawa1989** [comment]
close because this is addressed in #5662

**Pull request closed without merging**